drf_kit/views/viewsets.py

```diff
@@ -44,33 +46,52 @@ def get_serializer(self, *args, **kwargs):
         return super().get_serializer(*args, **kwargs)
 
     def get_response_serializer_class(self):
-        return self.serializer_detail_class or self.serializer_class
+        SERIALIZERS = {
+            'retrieve': self.serializer_detail_class,
+            'create': self.serializer_detail_class,
+            'update': self.serializer_detail_class,
+        }
+        action = self._get_action()
+
+        klass = SERIALIZERS.get(action)
+
+        if not klass:
+            klass = super().get_serializer_class()
+        return klass
 
     def get_response_serializer(self, obj, **kwargs):
         kwargs.update(self._get_serializer_extra_kwargs())
         return self.get_response_serializer_class()(obj, **kwargs)
 
     def get_queryset(self, *args, **kwargs):
-        queryset = None
-        verb = self.request.method.lower()
-
-        if verb == 'get' and self._is_request_to_detail_endpoint():
-            queryset = self.queryset_detail
-        elif verb == 'post':
-            queryset = self.queryset_create
-        elif verb == 'patch':
-            queryset = self.queryset_update or self.queryset_create
-
+        QUERYSETS = {
+            'retrieve': self.queryset_detail,
+            'create': self.queryset_create,
+            'update': self.queryset_update or self.queryset_create,
+        }
+        action = self._get_action()
+
+        queryset = QUERYSETS.get(action)
         if not queryset:
             queryset = super().get_queryset(*args, **kwargs)
         else:
             queryset = queryset.all()
         return queryset
 
-    def _is_request_to_detail_endpoint(self):
-        if hasattr(self, 'lookup_url_kwarg'):
-            lookup = self.lookup_url_kwarg or self.lookup_field
-        return lookup and lookup in self.kwargs
+    def _get_action(self):
+        def _is_request_to_detail_endpoint():
+            if hasattr(self, 'lookup_url_kwarg'):
+                lookup = self.lookup_url_kwarg or self.lookup_field
+            return bool(lookup and lookup in self.kwargs)
+
+        verb = self.request.method.lower()
+        if verb == 'get' and _is_request_to_detail_endpoint():
+            return 'retrieve'
+        if verb == 'post':
+            return 'create'
+        if verb == 'patch':
+            return 'update'
```

Comment:
why aren't you dealing with 'delete'?

Reply:
But it wasn't before, so okay

Reply:
The `delete` returns nothing (204) anyway.